frontend/public/components/utils/file-input.tsx

```diff
@@ -15,10 +18,14 @@ export class FileInput extends React.Component<FileInputProps, FileInputState> {
     this.props.onChange(event.target.value);
   }
   readFile(file) {
+    if (file.size > MAX_FILE_UPLOAD_SIZE) {
+      this.props.onChange('', '', fileSizeErrorMsg);
+      return;
+    }
     const reader = new FileReader();
     reader.onload = () => {
       const input = reader.result;
-      this.props.onChange(input, file.name);
+      this.props.onChange(input, file.name, '');
```

Comment:
It would be better if onChange took an object as an argument, then we could name the properties. It's not obvious when some of the values are empty what each parameter is for.